common/src/tap/checks/value_check.rs

```diff
@@ -0,0 +1,535 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use ::cost_model::CostModel;
+use anyhow::anyhow;
+use bigdecimal::ToPrimitive;
+use sqlx::{
+    postgres::{PgListener, PgNotification},
+    PgPool,
+};
+use std::{
+    collections::HashMap,
+    str::FromStr,
+    sync::{Arc, RwLock},
+};
+use thegraph_core::DeploymentId;
+use tracing::error;
+
+use tap_core::receipt::{
+    checks::{Check, CheckError, CheckResult},
+    state::Checking,
+    Context, ReceiptWithState,
+};
+
+use crate::cost_model;
+
+// we only accept receipts with minimal 1 wei grt
+const MINIMAL_VALUE: u128 = 1;
+
+pub struct AgoraQuery {
+    pub deployment_id: DeploymentId,
+    pub query: String,
+    pub variables: String,
+}
+
+type CostModelMap = Arc<RwLock<HashMap<DeploymentId, CostModel>>>;
+type GlobalModel = Arc<RwLock<Option<CostModel>>>;
+
+pub struct MinimumValue {
+    cost_model_map: CostModelMap,
+    global_model: GlobalModel,
+    watcher_cancel_token: tokio_util::sync::CancellationToken,
+}
+
+struct CostModelWatcher {
+    pgpool: PgPool,
+
```

Comment:
Is this newline meant to be here?

Reply:
I usually like to separate things inside structs like:
- Things related to database
- Things related to channels
- Things related to config

It's not a split inside the struct itself, but it helps when you want to abstract a whole portion into a separated struct.